modules/sqlite_helpers.py

```diff
@@ -118,10 +119,26 @@ def delete_url(sqlite_file: str, alias: str): #delete entry in the database from
 def maybe_delete_expired_url(sqlite_file, sqlite_row) -> bool: #returns True if url expired and deleted, otherwise False
     db = sqlite3.connect(sqlite_file)
     cursor = db.cursor()
+    current = datetime.now()
+    expiration_datetime_str = sqlite_row[4]
 
-    year_ago_date = datetime.now() - timedelta(days=365)
     result_datetime_str = sqlite_row[3].split(".")[0]  # Remove fractional seconds
     result_datetime = datetime.strptime(result_datetime_str, "%Y-%m-%d %H:%M:%S")
+
+    #copied code for condition that expiration date has passed
+    expiration_datetime = None
+    if sqlite_row[4] is not None:
+        expiration_datetime_str = sqlite_row[4].split(".")[0]
+        expiration_datetime_str = expiration_datetime_str.replace('T', ' ')
+        expiration_datetime = datetime.strptime(expiration_datetime_str, "%Y-%m-%d %H:%M:%S")
+
+        if current > expiration_datetime:
+            sql = "DELETE FROM urls WHERE alias = ?"
+            cursor.execute(sql, (sqlite_row[2], ))
+            db.commit()
+            return True
+
+    year_ago_date = current - timedelta(days=365)
```

Comment:
last thing, can we remove the logic to automatically delete urls that are a year old? from now on the user will have to specify an expiration date if they want it deleted

Reply:
should be done